src/module.cc

```diff
@@ -289,14 +308,37 @@ static void RestoreSignalHandler(int signo, struct sigaction* saved_sa) {
   sigaction(signo, saved_sa, nullptr);
 }
 
-// Raw signal handler for triggering a report - runs on an arbitrary thread
-static void SignalDump(int signo) {
+// Native signal handler for triggering a report on user signal - runs on an arbitrary thread
+static void SignalHandler(int signo, siginfo_t* info, void* void_context) {
   // Check atomic for report already pending, storing the signal number
   if (__sync_val_compare_and_swap(&report_signal, 0, signo) == 0) {
     uv_sem_post(&report_semaphore);  // Hand-off to watchdog thread
   }
 }
 
+// Native signal handler for triggering a report on a crash, runs on crashing thread
+static void CrashHandler(int signo, siginfo_t* info, void* void_context) {
+  // Remove node-report crash handlers in case we get a secondary crash
+  RestoreSignalHandler(signo, &saved_sigsegv_sa);
+  RestoreSignalHandler(signo, &saved_sigill_sa);
+  RestoreSignalHandler(signo, &saved_sigfpe_sa);
+  // Also remove the node-report user signal handler, if set
+  if (nodereport_events & NR_SIGNAL) {
+    RestoreSignalHandler(nodereport_signal, &saved_user_sa);
+  }
+  if ((info->si_pid == 0) || (info->si_pid == getpid())) {
```

Comment:
I'd drop the superfluous parens.  If you make a typo and write `info->si_pid = 0` instead, then gcc won't complain because it thinks the assignment is intentional.